calcom/cal.com
# fix: #16141 team name nullable

## What does this PR do?

Working with 16083 I found there are locations like "Google meet" which does not have `teamName` and it is set as `null`. If that is right, then the zod model is wrong. 

- Fixes #16141 
- Fixes CAL-none
Related with #16083 

## Mandatory Tasks (DO NOT REMOVE)


## How should this be tested?

- Are there environment variables that should be set? Default setup
- What are the minimal test data to have? run test
- What is expected (happy path) to have (input and output)? Editing a location of any Booking, and set a location without `teamName` is possible
- Any other important info that could help to test that PR
